README.md

```diff
@@ -0,0 +1,111 @@
+Custom Resource Status
+======================
+
+The purpose of this project is to provide some level of standardization and
+best-practices with respect to managing the status of custom resources. This project
+steals, err draws from:
+
+* [Cluster Version Operator (CVO)](https://github.com/openshift/cluster-version-operator)
+  that manages essential OpenShift operators.
+* [ClusterOperator Custom Resource](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md#what-should-an-operator-report-with-clusteroperator-custom-resource)
+  that exists for operators managed by CVO to communicate their status.
+* [openshift/library-go ClusterOperator status helpers](https://github.com/openshift/library-go/blob/master/pkg/config/clusteroperator/v1helpers/status.go)
+  that makes it easy to manage the status on a ClusterOperator resource.
+
+The goal here is to prescribe, without mandate, how to meaningfully populate the
+status of the Custom Resources your operator manages. Types, constants, and
+functions are provided for the following:
+
+* [Conditions](#Conditions)
+* [Object References](#Object-References)
+
+# Conditions
+
+Provides:
+
+* `Condition` type as specified in the [Kubernetes API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md)
+* `ConditionType` and generally useful constants for this type (ie. "Available",
+    "Progressing", "Degraded", and "Upgradeable")
+* Functions for setting, removing, finding, and evaluating conditions.
+
+To use, simply add `Conditions` to your Custom Resource Status struct like:
+
+```
+// ExampleAppStatus defines the observed state of ExampleApp
+type ExampleAppStatus struct {
+  ...
+  // conditions describes the state of the operator's reconciliation functionality.
+  // +patchMergeKey=type
+  // +patchStrategy=merge
+  // +optional
+  // Conditions is a list of conditions related to operator reconciliation
+  Conditions []conditions.Condition `json:"conditions,omitempty"  patchStrategy:"merge" patchMergeKey:"type"`
+}
+```
+
+Then, as appropriate in your Reconcile function, use
+`conditions.SetStatusConditions` like:
+
+```
+instance := &examplev1alpha1.ExampleApp{}
+err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+...handle err
+
+conditions.SetStatusCondition(&instance.Status.Conditions, conditions.Condition{
+  Type:   conditions.ConditionAvailable,
+  Status: corev1.ConditionFalse,
+  Reason: "ReconcileStarted",
+  Message: "Reconciling resource"
+})
+
+// Update the status
+err = r.client.Status().Update(context.TODO(), instance)
```

Comment:
I'm all for the granularity of exposing this, but it would be nice to combine this call with line 54 ```conditions.SetStatusCondition```.  Maybe a new function ```conditions.UpdateStatusCondition``` that does both set and the client call to update?

Reply:
I think this is something to consider. In the short term, my feeling is that the less that this library does...the harder it is to screw it up.

Reply:
Could be a reasonable addition if folks find value in it, but it would only save you one line of code. You'd also have to pass in the client to the new function. Right now I don't think there's any client code vendored in, which is somewhat nice.

Reply:
In the short term, my feeling is that the less that this library does the easier it is to get right.